content/help/ops/traffic-management/troubleshooting/index.md

```diff
@@ -4,8 +4,40 @@ description: Describes tools and techniques that can be used to root cause netwo
+
+* Determine where you’d like to access the application: from inside the same cluster or outside the cluster. If it’s outside the cluster, find out if port-forward is needed, if the right service type is configured for the service associated with the application, etc. If port-forward is needed, use the `kubectl port-forward` command to configure it.
+* Ping the node where the pod is running.
+* Check route table to make sure that there is a route to the pod IP: `ip route` or `route -n`.
+* Check to make sure that the return packet path is also properly configured. This involves checking the route tables, iptables, etc, in the node where the pod is running.
+* Check iptables to make sure that packets are not dropped.
+
+## Service discovery
+
+If service discovery fails, do the following:
+
+* Check if the service name associated with the application can be properly resolved: login to any other pod’s shell, use `nslookup` or `dig`, to find out if the name can be resolved. Otherwise, check to make sure that
+    * `kube-dns` (if used) is running properly
+    * `/etc/resolve.conf` is properly configured
+* Check if the port associated with the application has been opened in the pod: login to the pod or its namespace, and use `netstat` to find out.
+
+## Connection issues
+
+If your application runs a web service, you can `curl` the URLs that the application supports. If it returns errors such as ‘connection refused’, or empty content:
+
+* Check if `mTLS` has been enabled. And if it is, check the flags in your curl command are correct. Refer to [Mutual TLS](/docs/tasks/security/mutual-tls/#verifying-keys-and-certificates-installation).
+* Verify that the port used is correct and has been opened for communication.
+* Verify that the services that the application is calling are accessible and return successfully. This can be done by checking the sidecar logs of the application and the services with `kubectl log`.
+* Verify that iptables has been properly configured for the service associated with the application.
+* Verify that the application's envoy sidecar is up and running. Especially, verify that the sidecar has been successfully connected with istio Pilot. Refer to [Verifying connectivity to Istio Pilot](/help/ops/misc/#verifying-connectivity-to-istio-pilot).
+* Verify that the application's envoy sidecar has no significant errors in its log, especially errors with regard to mesh configuration.
+* Check Pilot's log and make sure that it doesn't contain errors related to the application.
+* Verify that envoy has been configured properly for the service associated with the application to work. Refer to [Debugging Envoy and Pilot](/help/ops/traffic-management/proxy-cmd/) on using `istioctl proxy-status` and `istioctl proxy-config` to do so.
```

Comment:
* Verify that you configured Envoy properly for the service associated with the application to work. Refer to [Debugging Envoy and Pilot](/help/ops/traffic-management/proxy-cmd/) for information on using `istioctl proxy-status` and `istioctl proxy-config` to configure Envoy correctly.